mistletoe/html_attributes_renderer.py

```diff
@@ -0,0 +1,256 @@
+"""
+HTML renderer for mistletoe.
+"""
+
+import html
+import re
+from itertools import chain
+from urllib.parse import quote
+from mistletoe import block_token
+from mistletoe import span_token
+from mistletoe.block_token import HTMLBlock
+from mistletoe.span_token import HTMLSpan
+from mistletoe.base_renderer import BaseRenderer
+
+
+class HTMLAttributesRenderer(BaseRenderer):
+    """
+    HTML renderer class.
+
+    See mistletoe.base_renderer module for more info.
+    """
+    def __init__(self, *extras):
+        """
+        Args:
+            extras (list): allows subclasses to add even more custom tokens.
+        """
+        self._suppress_ptag_stack = [False]
+        super().__init__(*chain((HTMLBlock, HTMLSpan), extras))
```

Comment:
This is probably where you'll want to insert your HTMLAttributes token class rather than in the BaseRenderer. As you can see, the other HTML tokens aren't referenced from the base renderer either.

On the other hand, if the HTMLAttributes is changed into something more generic then maybe the BaseRender is the right place after all. My preference would be to keep it purely optional i.e. not in the BaseRenderer. That way the performance hit is zero when the functionality isn't used, and will also be no unintended side effects when parsing markdown that happens to have HTMLAttributes-like tags in it.

Reply:
See (https://github.com/miyuchina/mistletoe/pull/170/commits/e68273d0395889f17b8bfc46a49a235555320bc6)